cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/Capacity.java

```diff
@@ -0,0 +1,119 @@
+    public static final int DEFAULT_BROKER_CPU_CAPACITY    = 100;     // as a percentage (0-100)
+    public static final int DEFAULT_BROKER_NW_IN_CAPACITY  = 10000;   // in KB/s
+    public static final int DEFAULT_BROKER_NW_OUT_CAPACITY = 10000;   // in KB/s
+
+    private Integer disk;
+    private Integer cpu;
+    private Integer nwIn;
+    private Integer nwOut;
+
+    public Capacity(KafkaSpec spec) {
+        CruiseControlBrokerCapacity bc = spec.getCruiseControl().getCapacity();
+
+        if (bc != null) {
+            this.disk = bc.getDisk() != null ? bc.getDisk() : generateDiskCapacity(spec.getKafka().getStorage());
+            this.cpu = bc.getCpu() != null ? bc.getCpu() : DEFAULT_BROKER_CPU_CAPACITY;
+            this.nwIn = bc.getNwIn() != null ? bc.getNwIn() : DEFAULT_BROKER_NW_IN_CAPACITY;
+            this.nwOut = bc.getNwOut() != null ? bc.getNwOut() : DEFAULT_BROKER_NW_OUT_CAPACITY;
+        } else {
+            this.disk = generateDiskCapacity(spec.getKafka().getStorage());
+            this.cpu = DEFAULT_BROKER_CPU_CAPACITY;
+            this.nwIn = DEFAULT_BROKER_NW_IN_CAPACITY;
+            this.nwOut = DEFAULT_BROKER_NW_OUT_CAPACITY;
+        }
+    }
+
+    /**
+     * Generate disk capacity configuration from the supplied storage configuration
+     *
+     * @param storage Storage configuration for Kafka cluster
+     * @return Disk capacity configuration as a String
+     */
+    public static Integer generateDiskCapacity(Storage storage) {
+        if (storage instanceof PersistentClaimStorage) {
+            return getSizeInMb(((PersistentClaimStorage) storage).getSize());
+        } else if (storage instanceof EphemeralStorage) {
+            if (((EphemeralStorage) storage).getSizeLimit() != null) {
+                return getSizeInMb(((EphemeralStorage) storage).getSizeLimit());
+            } else {
+                return DEFAULT_BROKER_DISK_CAPACITY;
+            }
+        } else if (storage instanceof JbodStorage) {
+            List<SingleVolumeStorage> volumeList = ((JbodStorage) storage).getVolumes();
+            int size = 0;
+            for (SingleVolumeStorage volume : volumeList) {
+                size += generateDiskCapacity(volume);
+            }
+            return size;
+        } else {
+            throw new IllegalStateException("The declared storage '" + storage.getType() + "' is not supported");
+        }
+    }
+
+    /*
+     * Parse a K8S-style representation of a disk size, such as {@code 100Gi},
+     * into the equivalent number of megabytes represented as a Integer.
+     *
+     * @param size The String representation of the volume size.
+     * @return The equivalent number of Megabytes.
+     */
+    public static Integer getSizeInMb(String size) {
+        long factor = 1_000L;
+        if (size.charAt(size.length() - 1) == 'i') {
+            factor = 1_024L;
+        }
+        return Math.toIntExact(parseMemory(size) / (factor * factor));
+    }
```

Comment:
can you verify with @tombentley if we already have something around about this in the code?

Reply:
@tomncooper check out `Quantites` to see if it's of any relevance.

Reply:
I think Tom meant to tag @ppatierno  here. I was thinking of creating a function in the Quantities class that would allow for converting the byte value returned from parseMemory to any unit KB, MB, GB ect but wasn't sure if it would be worth it. If you think it is, I will take a stab at it!

Reply:
Added a commit of how this could be done alternatively, let me know what you think!